src/Common/com/bioxx/tfc/Handlers/CraftingHandler.java

```diff
@@ -205,38 +204,14 @@ else if(itemstack.getItem() == TFCItems.PotterySmallVessel && itemstack.getItemD
 					for(int id : ids)
 					{
 						String name = OreDictionary.getOreName(id);
-						if(name.equals("dyeWhite"))
-							c = EntitySheep.fleeceColorTable[0];
-						else if(name.equals("dyeOrange"))
-							c = EntitySheep.fleeceColorTable[1];
-						else if(name.equals("dyeMagenta"))
-							c = EntitySheep.fleeceColorTable[2];
-						else if(name.equals("dyeLightBlue"))
-							c = EntitySheep.fleeceColorTable[3];
-						else if(name.equals("dyeYellow"))
-							c = EntitySheep.fleeceColorTable[4];
-						else if(name.equals("dyeLime"))
-							c = EntitySheep.fleeceColorTable[5];
-						else if(name.equals("dyePink"))
-							c = EntitySheep.fleeceColorTable[6];
-						else if(name.equals("dyeGray"))
-							c = EntitySheep.fleeceColorTable[7];
-						else if(name.equals("dyeLightGray"))
-							c = EntitySheep.fleeceColorTable[8];
-						else if(name.equals("dyeCyan"))
-							c = EntitySheep.fleeceColorTable[9];
-						else if(name.equals("dyePurple"))
-							c = EntitySheep.fleeceColorTable[10];
-						else if(name.equals("dyeBlue"))
-							c = EntitySheep.fleeceColorTable[11];
-						else if(name.equals("dyeBrown"))
-							c = EntitySheep.fleeceColorTable[12];
-						else if(name.equals("dyeGreen"))
-							c = EntitySheep.fleeceColorTable[13];
-						else if(name.equals("dyeRed"))
-							c = EntitySheep.fleeceColorTable[14];
-						else if(name.equals("dyeBlack"))
-							c = EntitySheep.fleeceColorTable[15];
+						for(int j = 1; j < Global.DYE_NAMES.length; j++)

```

Comment:
This should have been j=0. If you start with j=1, then you can't dye your vessels white.